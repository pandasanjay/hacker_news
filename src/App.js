import React from 'react'
import PropTypes from 'prop-types'
import { renderRoutes } from 'react-router-config'
import './styles/style.scss'
import ErrorBoundary from './components/ErrorBoundary'

export const App = ({ route }) => {
    return (
        <>
            <div>
                <ErrorBoundary>{renderRoutes(route.routes)}</ErrorBoundary>
            </div>
        </>
    )
}
App.propTypes = {
    route: PropTypes.object,
}
export default {
    component: App,
}
