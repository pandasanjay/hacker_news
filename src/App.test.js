import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'
import configureStore from 'redux-mock-store'
import { App } from './App'
import Routers from './Routes'
const mockStore = configureStore([])
describe('App', () => {
    let store

    beforeEach(() => {
        store = mockStore({
            appState: {
                name: 'hello',
            },
        })
    })
    it('should render app and check snapshot', () => {
        const { asFragment } = render(
            <Provider store={store}>
                <Router>
                    <App route={Routers[0]} />
                </Router>
            </Provider>
        )
        expect(asFragment()).toMatchSnapshot()
    })
})
