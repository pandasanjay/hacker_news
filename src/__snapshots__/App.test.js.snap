// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render app and check snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      My story
    </div>
  </div>
</DocumentFragment>
`;
